mail/mailer.go

```diff
@@ -32,12 +38,31 @@ func New(server, port, username, password string) MailerImpl {
 		Port:   port,
 		Auth:   auth,
 		From:   username,
+		clk:    clock.Default(),
+	}
+}
+
+func (m *MailerImpl) generateMessage(to []string, subject, body string) []byte {
+	now := m.clk.Now().UTC()
+	rand.Seed(int64(now.Nanosecond()))
```

Comment:
This actually needs to be cryptographically random. Please use crypto/rand and don't seed it by time.

If you need an interface for testing that will use math/rand under the hood in tests, write one.